src/components/PersonLink.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Link, useLocation } from 'react-router-dom';
+
+interface PersonLinkProps {
+  name: string;
+  sex: string;
+  slug: string;
+}
+
+export const PersonLink: React.FC<PersonLinkProps> = ({ name, sex, slug }) => {
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const currentCentury = searchParams.getAll('centuries');
+  const currentSort = searchParams.get('sort');
+  const currentOrder = searchParams.get('order');
+
+  const newSearchParams = new URLSearchParams();
+
+  if (currentCentury) {
+    currentCentury.forEach(century => {
+      newSearchParams.append('centuries', century);
+    });
+  }
```

Comment:
The check for `currentCentury` should ensure it is not an empty array before appending to `newSearchParams`. Consider checking `currentCentury.length` to avoid unnecessary operations when the array is empty.